src/ConnectionFX3/ConnectionFX3.cpp

```diff
@@ -642,17 +635,10 @@ bool ConnectionFX3::WaitForSending(int contextHandle, unsigned int timeout_ms)
 	status = contextsToSend[contextHandle].EndPt->WaitForXfer(contextsToSend[contextHandle].inOvLap, timeout_ms);
 	return status;
 #   else
-    auto t1 = chrono::high_resolution_clock::now();
-    auto t2 = t1;
-
+    //blocking not to waste CPU
     std::unique_lock<std::mutex> lck(contextsToSend[contextHandle].transferLock);
-    while(contextsToSend[contextHandle].done.load() == false && std::chrono::duration_cast<std::chrono::milliseconds>(t2 - t1).count() < timeout_ms)
-    {
-        //blocking not to waste CPU
-        contextsToSend[contextHandle].cv.wait_for(lck, chrono::milliseconds(timeout_ms));
-        t2 = chrono::high_resolution_clock::now();
-    }
-	return contextsToSend[contextHandle].done == true;
+    contextsToSend[contextHandle].cv.wait_for(lck, chrono::milliseconds(timeout_ms), [&](){return contextsToSend[contextHandle].done.load();});
+    return contextsToSend[contextHandle].done == true;
```

Comment:
On a timeout, while unlikely, contextsToSend[contextHandle].done value could have changed, since you are not loading it anymore.